src/socket_base.cpp

```diff
@@ -1414,6 +1414,7 @@ void zmq::socket_base_t::update_pipe_options(int option_)
         for (pipes_t::size_type i = 0; i != pipes.size(); ++i)
         {
             pipes[i]->set_hwms(options.rcvhwm, options.sndhwm);
+            pipes[i]->set_peer_hwms(options.sndhwm, options.rcvhwm);
```

Comment:
I think the command is the best option. I'd need to double check as can't remember exactly, but I think even with other transports this might be unsafe as the peer pipe is used by the I/O thread, unless I'm mistaken